### PR TITLE
New test adding a bug to the pinboard

### PR DESCRIPTION
**Pull request by rbillings**
Fixes issue #47 
*Note: two save methods were used as the primary save button was not entering bug data into the job details panel.

### CODE REVIEW

**Comment by rbillings** [comment]
Rebased

**Comment by stephendonner** [comment]
Hmm, this fails nearly 100% of the time for me, sadly:

`(treeherder-tests) sdonners-MacBook-Pro:treeherder-tests sdonner$ py.test --base-url https://treeherder.allizom.org --driver Firefox tests/
__init__.py                    test_expanding_group_count.py  test_switch_repo.py
__init__.pyc                   test_get_next_results.py       test_treeherder_menu.py
__pycache__/                   test_login.py                  test_unclassified_job.py
conftest.py                    test_pin_job.py                test_view_single_result.py
test_add_new_jobs.py           test_range.py                  
(treeherder-tests) sdonners-MacBook-Pro:treeherder-tests sdonner$ py.test --base-url https://treeherder.allizom.org --driver Firefox tests/test_pin_job.py 
============================================ test session starts =============================================
platform darwin -- Python 2.7.11, pytest-3.0.0, py-1.4.31, pluggy-0.3.1
baseurl: https://treeherder.allizom.org
driver: Firefox
sensitiveurl: mozilla\.org
rootdir: /Users/sdonner/treeherder-tests, inifile: setup.cfg
plugins: base-url-1.1.0, html-1.10.0, rerunfailures-0.5, selenium-1.3.1, variables-1.4, xdist-1.15.0
collected 5 items 

tests/test_pin_job.py ....F

================================================== FAILURES ==================================================
_______________________________________________ test_pin_a_bug _______________________________________________

base_url = 'https://treeherder.allizom.org'
selenium = <selenium.webdriver.firefox.webdriver.WebDriver (session="28065306-e4ed-3b4b-a4fe-9dbfbda24f79")>
new_user = {'email': 'olivares625797@personatestuser.org', 'name': 'olivares625797', 'password': 'whqL6DroIOCvcynB', 'url': 'http://www.mozilla.org/', ...}

    def test_pin_a_bug(base_url, selenium, new_user):
        """Open treeherder, log in, select unclassified job, pin job, add a bug, save and verify"""
        page = TreeherderPage(selenium, base_url).open()
        page.header.login(new_user['email'], new_user['password'])
        page.open_next_unclassified_failure()
        assert not page.job_details.is_job_bug_visible
        page.pin_using_spacebar()
    
        bug_id = 1164485
        page.pinboard.add_bug_to_pinned_job(bug_id)
        page.pinboard.save_bug_to_pinboard()
>       assert page.job_details.is_job_bug_visible
E       assert False
E        +  where False = <pages.treeherder.JobDetails object at 0x1041a76d0>.is_job_bug_visible
E        +    where <pages.treeherder.JobDetails object at 0x1041a76d0> = <pages.treeherder.TreeherderPage object at 0x1041a7f50>.job_details

tests/test_pin_job.py:59: AssertionError
---------------------------------------------- pytest-selenium -----------------------------------------------
URL: https://treeherder.allizom.org/#/jobs?repo=mozilla-inbound&selectedJob=32264669
==================================== 1 failed, 4 passed in 42.80 seconds =====================================`

**Comment by rbillings** [comment]
So kooky! You find good stuff @stephendonner - I'll try to repro on my end.

**Comment by rbillings** [comment]
Rebased and ready for review @mozilla/web-qa-sorcerers

File: tests/test_pin_job.py [hunk 1]
```diff
@@ -43,3 +43,20 @@ def test_pin_all_jobs(base_url, selenium):
     page = TreeherderPage(selenium, base_url).open()
     page.result_sets[0].pin_all_jobs()
     assert 0 < len(page.pinboard.jobs) <= 500
+
+
+def test_pin_a_bug(base_url, selenium, new_user):
+    """Open treeherder, log in, select unclassified job, pin job, add a bug, save and verify"""
+    page = TreeherderPage(selenium, base_url).open()
+    page.header.login(new_user['email'], new_user['password'])
```
Comment (stephendonner) [inline tests/test_pin_job.py]: I'm seeing a failure where the bidpom/Persona-login window hasn't-yet loaded - can a conditional wait be added to make this more robust?

`________________________________ test_pin_a_bug ________________________________
[gw7] linux2 -- Python 2.7.9 /var/lib/jenkins/jobs/treeherder.adhoc/workspace/.venv/bin/python2.7
base_url = 'https://treeherder.allizom.org'
selenium = <selenium.webdriver.remote.webdriver.WebDriver (session="84d3eeeefa2c4b0ea063b5c84ffa7ff0")>
new_user = {'email': 'charlie633686@personatestuser.org', 'name': 'charlie633686', 'password': 'jIUNgVygizhS4hVA', 'url': 'http://www.mozilla.org/', ...}

    def test_pin_a_bug(base_url, selenium, new_user):
        """Open treeherder, log in, select unclassified job, pin job, add a bug, save and verify"""
        page = TreeherderPage(selenium, base_url).open()
>       page.header.login(new_user['email'], new_user['password'])

tests/test_pin_job.py:51: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pages/base.py:35: in login
    browser_id.sign_in(email, password)
.venv/lib/python2.7/site-packages/bidpom/bidpom.py:20: in sign_in
    sign_in = SignIn(self.selenium, self.timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <bidpom.pages.sign_in.SignIn object at 0x7faa7a130510>
selenium = <selenium.webdriver.remote.webdriver.WebDriver (session="84d3eeeefa2c4b0ea063b5c84ffa7ff0")>
timeout = 10

    def __init__(self, selenium, timeout=10):
        super(SignIn, self).__init__(selenium, timeout)
    
        if not self.selenium.title == self._page_title:
            for handle in self.selenium.window_handles:
                self.selenium.switch_to_window(handle)
                self.wait.until(lambda s: s.title)
                if self.selenium.title == self._page_title:
                    self._sign_in_window_handle = handle
                    break
            else:
>               raise Exception('Popup has not loaded')
E               Exception: Popup has not loaded

.venv/lib/python2.7/site-packages/bidpom/pages/sign_in.py:50: Exception
------------------------------- pytest-selenium --------------------------------
URL: https://treeherder.allizom.org/admin.html#/profiles
SauceLabs Job: http://saucelabs.com/jobs/84d3eeeefa2c4b0ea063b5c84ffa7ff0
________________________ test_that_a_new_user_can_login ________________________
[gw4] linux2 -- Python 2.7.9 /var/lib/jenkins/jobs/treeherder.adhoc/workspace/.venv/bin/python2.7
base_url = 'https://treeherder.allizom.org'
selenium = <selenium.webdriver.remote.webdriver.WebDriver (session="dca598fcea74467382f170fa185298af")>
new_user = {'email': 'roy633687@personatestuser.org', 'name': 'roy633687', 'password': 'Dm9n5qEOUHjVFBgr', 'url': 'http://www.mozilla.org/', ...}

    @pytest.mark.nondestructive
    def test_that_a_new_user_can_login(base_url, selenium, new_user):
        page = TreeherderPage(selenium, base_url).open()
>       page.header.login(new_user['email'], new_user['password'])

tests/test_login.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
pages/base.py:35: in login
    browser_id.sign_in(email, password)
.venv/lib/python2.7/site-packages/bidpom/bidpom.py:20: in sign_in
    sign_in = SignIn(self.selenium, self.timeout)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <bidpom.pages.sign_in.SignIn object at 0x7f43e8b4dc50>
selenium = <selenium.webdriver.remote.webdriver.WebDriver (session="dca598fcea74467382f170fa185298af")>
timeout = 10

    def __init__(self, selenium, timeout=10):
        super(SignIn, self).__init__(selenium, timeout)
    
        if not self.selenium.title == self._page_title:
            for handle in self.selenium.window_handles:
                self.selenium.switch_to_window(handle)
                self.wait.until(lambda s: s.title)
                if self.selenium.title == self._page_title:
                    self._sign_in_window_handle = handle
                    break
            else:
>               raise Exception('Popup has not loaded')
E               Exception: Popup has not loaded

.venv/lib/python2.7/site-packages/bidpom/pages/sign_in.py:50: Exception`

**Comment by rbillings** [comment]
Added an assert for login, @stephendonner. Ready for review, @mozilla/web-qa-sorcerers

**Comment by stephendonner** [comment]
Now I'm getting a new failure where |message| is null/empty string:

http://pastebin.com/tTAHixjv

**Comment by rbillings** [comment]
@stephendonner - I am unable to repro that failure locally or on Sauce, would you please try re-running the test?

**Comment by stephendonner** [comment]
@rbillings now it's complaining that the popup never loaded:

https://pastebin.mozilla.org/8913628

**Comment by rbillings** [comment]
I'm sorry to hear that @stephendonner. I was unable to repro locally or on Sauce- and bidpom errors don't relate to the code change.  :(

**Comment by m8ttyB** [comment]
r+

**Comment by m8ttyB** [comment]
amazing

**Comment by rbillings** [comment]
I'm going to go ahead and merge with an R+ and the green buttons. I'll re-open/fix if need be.

**Pull request merged**